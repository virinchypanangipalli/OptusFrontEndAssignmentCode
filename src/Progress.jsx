import React from "react";

const Progress = ({ bar }) => {
  return (
    <div className="progress">
      <div
        className="progress-done"
        style={{ opacity: 1, width: `${bar.bar}%`, background: bar.background }}
      >
        <span style={{ color: bar.color }}>{bar.bar}%</span>
      </div>
    </div>
  );
};

export default Progress;
