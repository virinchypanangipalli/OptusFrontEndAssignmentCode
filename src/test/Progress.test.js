import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Progress from "../Progress";
configure({ adapter: new Adapter() });

describe("Testing Progress Bar Rendering ", () => {
  let wrapper;
  beforeAll(() => {
    let obj = { bar: 0, background: "lightblue" };
    wrapper = shallow(<Progress bar={obj} />);
    // wrapper = shallow(<ProgressBar getbars={getbars} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  test("It Should render without errors ", () => {
    expect(wrapper.find("span").text()).toContain(0);
  });
});
