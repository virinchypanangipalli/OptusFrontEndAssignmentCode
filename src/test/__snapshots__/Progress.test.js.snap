// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <div
    className="progress"
  >
    <div
      className="progress-done"
      style={
        Object {
          "background": "lightblue",
          "opacity": 1,
          "width": "0%",
        }
      }
    >
      <span
        style={
          Object {
            "color": undefined,
          }
        }
      >
        0
        %
      </span>
    </div>
  </div>,
]
`;
